### ISSUE TITLE
Fix PSQL Support 

### ISSUE THREAD

**Issue by apavlo** [cmu-db/peloton#223]
There is something wrong with our new networking code where we can't use JDBC or PSQL at the same time. I'm not sure what the problem is, but this is obviously bad. We need to fix this.

**Comment by sid1607** [comment]
To clarify, JDBC and PSQL can't be used concurrently but work properly when they are running on their separately?

**Comment by apavlo** [comment]
Yes, this is my understanding.

@jarulraj This is correct, right?

**Comment by jarulraj** [comment]
I discussed this with @sid1607.

**Comment by sid1607** [comment]
Fixes
------

1) PSQL and OLTPBench can both work using AF_INET sockets. The terminal README has been updated - https://github.com/cmu-db/peloton/wiki/Terminal

2) wire/protocol.cpp had a bug where it would assume every query was an empty query and did not print any output on PSQL. This has been fixed by 86360f01793e3bd4d829e0c70e534b6f09137244

Open Issues
----------------
1) The Query Executor only returns the first tuple that was inserted as the result for ``SELECT *`` queries on PSQL.

2) Join queries on PSQL crash by throwing
``` 
throwing an instance of 'std::logic_error'
  what():  basic_string::_S_construct null not valid
```
